runtime/tests/checker/reference_test.go

```diff
@@ -213,12 +286,59 @@ func TestCheckReferenceExpressionWithInterfaceResultType(t *testing.T) {
 	})
 }
 
+func TestCheckReferenceExpressionWithRdAnyResultType(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("resource", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+          resource R {}
+
+          let r <- create R()
+          let ref = &r as &AnyResource
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("struct", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+          struct S {}
+
+          let s = S()
+          let ref = &s as &AnyStruct
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("non-composite", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+          let i = 1
+          let ref = &i as &AnyStruct
```

Comment:
Hmm, it seems a bit odd to me we can cast a reference to an `Int` to reference to `AnyStruct`

Reply:
`Int` is a subtype of `AnySruct`. I'll try to document this better through #237